flink-clients/src/main/java/org/apache/flink/client/cli/CliFrontend.java

```diff
@@ -892,15 +895,20 @@ private JobID parseJobId(String jobIdString) throws CliArgsException {
 	 * @throws FlinkException if something goes wrong
 	 */
 	private <ClusterID> void runClusterAction(CustomCommandLine activeCommandLine, CommandLine commandLine, ClusterAction<ClusterID> clusterAction) throws FlinkException {
-		final Configuration executorConfig = activeCommandLine.applyCommandLineOptionsToConfiguration(commandLine);
-		final ClusterClientFactory<ClusterID> clusterClientFactory = clusterClientServiceLoader.getClusterClientFactory(executorConfig);
+		final Configuration effectiveConfiguration = new Configuration(configuration);
+		final Configuration commandLineConfiguration = activeCommandLine.toConfiguration(commandLine);
+		effectiveConfiguration.addAll(commandLineConfiguration);
 
-		final ClusterID clusterId = clusterClientFactory.getClusterId(executorConfig);
+		LOG.debug("Effective configuration: {}", effectiveConfiguration);
```

Comment:
Given that the `env` itself can add config options later, maybe the message here could change to reflect that this is the `effective configuration at the CLI client` or `after the addition of the CLI options`. Although a bit esoteric, this is a debug message so I think it is ok. If we want the actual effective config I think we should go in the `env.executeAsync()`.

Reply:
Will add that. 👌